### PR TITLE
Optimizations and changes

### LINKED ISSUE #36: Fails to save run if the new model instance has same error value as a previous one

### LINKED ISSUE #37: Data from different cells are mixed on train

### LINKED ISSUE #38: XGBoost and randomforest are being recreated on training

### LINKED ISSUE #39: Refactor models
Refactor modelI and models to use a better data format

### LINKED ISSUE #40: Refactor training
Change trainer to a better and more efficient approach

### LINKED ISSUE #41: Refactor inferece
same as #40

### LINKED ISSUE #42: Remove XGBoost and RandomForest
XGBoost and randomforest shouldn't be used here.

### PR DESCRIPTION
**Pull request by ThiagoAVicente**
## Summary
This pr simplifies the logic for inference and train jobs. Also add input normalization to reduce loss. 
> LSTM had MSE=3000 after 9 runs and now it has 8 after 1 run

## Key changes
- Add ANN ( artificial neural network), it is a faster and "dumber" alternative to LSTM that'll help on presentation ig
- Add normalization, normalization is done on model train and then it is stored as a artifact. Artifacts are loaded and used on inference process
- Add method to `MLInterface` to load artifacts
- Removed **XGBoost** and **RandomForest** from registered models.
> This because I'm lazy and too much effort is required to make the system work with NNs and models like XGBoost at the same time. 

## New rules
- To request analytics from a model, a previous train job must run
- Infer and train now contains **cell_index** so the model, *maybe*, can learn how to behave with different cells.

closes #42 
closes #41 
closes #39 
closes #40 
closes #38 
closes #37 
closes #36

### CODE REVIEW

**Review by Alexandre-A — APPROVED** [review]
Nice work!

**Review by alxmra — APPROVED** [review]

**Review by alxmra — APPROVED** [review]
